network-protection/network-protection-api/src/main/java/com/duckduckgo/networkprotection/api/NetworkProtectionState.kt

```diff
@@ -17,6 +17,13 @@
 package com.duckduckgo.networkprotection.api
 
 interface NetworkProtectionState {
+
+    /**
+     * This method returns whether the user is authenticated and has obtained a token.
+     * @return Returns `true` if the user is authenticated, `false` otherwise
+     */
+    fun isAuthenticated(): Boolean
```

Comment:
I was writing this method too for other things. I was calling it `isOnboarded()` similar to what ApptP has.

The implementation in `NetworkProtectionStateImpl` was the following
```kotlin
    override suspend fun isOnboarded(): Boolean = withContext(dispatcherProvider.io()) {
        return@withContext netpCohortStore.cohortLocalDate != null
    }
```

Reply:
@aitorvs I didn't think of doing the same mainly because we don't have any onboarding on NetP but i guess onboarding can mean other things (the new invite code flow could be considered an onboarding). Will update for consistency.